crates/matrix-sdk-crypto-ffi/src/machine.rs

```diff
@@ -134,28 +135,51 @@ impl OlmMachine {
     }
 
     /// Get a cross signing user identity for the given user ID.
-    pub fn get_identity(&self, user_id: &str) -> Result<Option<UserIdentity>, CryptoStoreError> {
+    ///
+    /// # Arguments
+    ///
+    /// * `user_id` - The unique id of the user that the identity belongs to
+    ///
+    /// * `timeout` - The amount of time we should wait before returning if the
```

Comment:
Let's be specific here:
```suggestion
    /// * `timeout` - The time in seconds we should wait before returning if the
```

Also... we handle `0` in a special manner - we should document that, too. Or, better, can't we make that more explicit by making this an `Option<u32>` itself? Is there any pattern where we might to pass an explicit timeout of `0`? Either way, I think the `None` is a much cleaner indicator for "wait until forever" than `0` is...

Reply:
We convert a `0` to a `None`, which means "don't wait at all". If you still think that `0` is a poor indicator of "don't wait at all" we can make the `u32` optional, but I don't see much point to doing it.

Reply:
so, "don't wait at all" means err immediately if not already there? That sounds like a fair way of understanding it, but in that case I'd understand a `None` in the API below as a _wait for ever_... which it isn't though, right?

Either way, this isn't super intuitive and either choice is fine with me but should be documented very precisely..

Reply:
Don't wait at all means, we're not waiting for anything, it's a simple getter that fetches something from the store. Passing a `None` makes it work like it worked before we had any "wait" mechanism.

We added to the getters in the `OlmMachine` this line:
```rust
self.wait_if_user_pending(user_id, timeout).await;
```

If we inspect the method:

```rust
impl OlmMachine {
    async fn wait_if_user_pending(&self, user_id: &UserId, timeout: Option<Duration>) {
        if let Some(timeout) = timeout {
            let listener = self.identity_manager.listen_for_received_queries();

            let _ = listener.wait_if_user_pending(timeout, user_id).await;
        }
    }
}
    ```

We can see that passing a `None` turns that method into a noop. There is no way to wait forever since that seems rather pointless. In fact I'm not entirely convinced that waiting for this is useful at all. I'm only adding this to make the EA unit tests happy. That's also the reason why the `matrix-sdk` crate doesn't use this.

Reply:
understood. let's a docs comments on the behaviour of `0` and that's all fine with me.

Reply:
Added in https://github.com/matrix-org/matrix-rust-sdk/pull/736/commits/43791ed966782dc4c66662016adcd55b43c31d9b